internal/common/constants.go

```diff
@@ -0,0 +1,160 @@
+	// should be reflected on the pods.
+	ConfigMapCheckSumKey = "checksum/etcd-configmap"
+)
+
+// Constants for container names
+const (
+	// EtcdContainerName is the name of the etcd container.
+	EtcdContainerName = "etcd"
+	// EtcdBackupRestoreContainerName is the name of the backup-restore container.
+	EtcdBackupRestoreContainerName = "backup-restore"
+	// ChangePermissionsInitContainerName is the name of the change permissions init container.
+	ChangePermissionsInitContainerName = "change-permissions"
+	// ChangeBackupBucketPermissionsInitContainerName is the name of the change backup bucket permissions init container.
+	ChangeBackupBucketPermissionsInitContainerName = "change-backup-bucket-permissions"
+)
+
+// Constants for volume names
+const (
+	// EtcdCAVolumeName is the name of the volume that contains the CA certificate bundle and CA certificate key used to sign certificates for client communication.
+	EtcdCAVolumeName = "etcd-ca"
+	// EtcdServerTLSVolumeName is the name of the volume that contains the server certificate-key pair used to set up the etcd server and etcd-wrapper HTTP server.
+	EtcdServerTLSVolumeName = "etcd-server-tls"
+	// EtcdClientTLSVolumeName is the name of the volume that contains the client certificate-key pair used by the client to communicate to the etcd server and etcd-wrapper HTTP server.
+	EtcdClientTLSVolumeName = "etcd-client-tls"
+	// EtcdPeerCAVolumeName is the name of the volume that contains the CA certificate bundle and CA certificate key used to sign certificates for peer communication.
+	EtcdPeerCAVolumeName = "etcd-peer-ca"
+	// EtcdPeerServerTLSVolumeName is the name of the volume that contains the server certificate-key pair used to set up the peer server.
+	EtcdPeerServerTLSVolumeName = "etcd-peer-server-tls"
+	// BackupRestoreCAVolumeName is the name of the volume that contains the CA certificate bundle and CA certificate key used to sign certificates for backup-restore communication.
+	BackupRestoreCAVolumeName = "backup-restore-ca"
+	// BackupRestoreServerTLSVolumeName is the name of the volume that contains the server certificate-key pair used to set up the backup-restore server.
+	BackupRestoreServerTLSVolumeName = "backup-restore-server-tls"
+	// BackupRestoreClientTLSVolumeName is the name of the volume that contains the client certificate-key pair used by the client to communicate to the backup-restore server.
+	BackupRestoreClientTLSVolumeName = "backup-restore-client-tls"
+
+	// EtcdConfigVolumeName is the name of the volume that contains the etcd configuration file.
+	EtcdConfigVolumeName = "etcd-config-file"
+	// LocalBackupVolumeName is the name of the volume that contains the local backup.
+	LocalBackupVolumeName = "local-backup"
+	// ProviderBackupSecretVolumeName is the name of the volume that contains the provider backup secret.
+	ProviderBackupSecretVolumeName = "etcd-backup-secret"
+)
+
```

Comment:
Although it's unlikely that permissions would be changed - having what the permissions `0640` do makes it clear to the developer what permissions are provided as default.
A potential change in the future will also be easy if a `const` is used instead of hardcoding.
Please expand on the comment.

```suggestion

// OwnerReadWriteGroupReadPermissions is the file permissions used for volumes
const OwnerReadWriteGroupReadPermissions int32 = 0640

```

Reply:
Done, thanks for pointing this out.